src/python_testing/TC_TSTAT_4_2.py

```diff
@@ -461,8 +465,62 @@ async def test_TC_TSTAT_4_2(self):
             # Roll back
             await self.send_atomic_request_rollback_command()
 
-        # TODO: Add tests for the total number of Presets exceeds the NumberOfPresets supported. Also Add tests for adding presets with preset scenario not present in PresetTypes.
+        self.step("17")
+        if self.pics_guard(self.check_pics("TSTAT.S.F08") and self.check_pics("TSTAT.S.A0050") and self.check_pics("TSTAT.S.Cfe.Rsp")):
+            test_presets = new_presets_with_handle.copy()
+            test_presets.append(cluster.Structs.PresetStruct(presetHandle=NullValue, presetScenario=9,
+                                                             name="Wake", coolingSetpoint=2500, heatingSetpoint=1700, builtIn=False))
+
+            # Send the AtomicRequest begin command
+            await self.send_atomic_request_begin_command()
+
+            await self.write_presets(endpoint=endpoint, presets=test_presets, expected_status=Status.ConstraintError)
+
+            # Clear state for next test.
+            await self.send_atomic_request_rollback_command()
+
+        self.step("18")
+        if self.pics_guard(self.check_pics("TSTAT.S.F08") and self.check_pics("TSTAT.S.A0050") and self.check_pics("TSTAT.S.Cfe.Rsp")):
+
+            # Read the numberOfPresets supported.
+            numberOfPresetsSupported = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=cluster.Attributes.NumberOfPresets)
+
+            # Read the PresetTypes to get the preset scenarios to build the Presets list.
+            presetTypes = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=cluster.Attributes.PresetTypes)
+
+            # Read the Presets to copy the existing presets into our testPresets list below.
+            presets = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=cluster.Attributes.Presets)
 
+            # Since we currently support one preset of each type, run the test only if we have enough preset types to create a preset list whose length exceeds the
```

Comment:
i modified the comment to say sdk instead of "we". Also there is already an issue to support multiple presets of each type in the sdk - https://github.com/project-chip/connectedhomeip/issues/34556. will fix this as part of that. added a comment in the issue to track this.